### PR TITLE
migrate to null safety

### PR DESCRIPTION
**Pull request by zoliabraham**
Migrated to null safety and updated depricated android files in example project

### CODE REVIEW

File: example/lib/component/password_field.dart [hunk 1]
```diff
@@ -32,19 +30,11 @@ class _PasswordFieldState extends State<PasswordField> {
       keyboardType: TextInputType.text,
       obscureText: true,
       validator: validatePassword,
-      autovalidate: _isPasswordDirty,
+      autovalidateMode: AutovalidateMode.always,
```
Comment (orevial) [inline example/lib/component/password_field.dart:33]: I think the equivalent of the `_isPasswordDirty` boolean is `autovalidateMode: AutovalidateMode.onUserInteraction`.

**Review by orevial — CHANGES_REQUESTED** [review]
Awesome, I was going to propose a similar PR, thank you 🙏  🎉

File: example/lib/component/password_field.dart (hunk 1 above)
Comment (zoliabraham) [reply example/lib/component/password_field.dart:33]: You are right. Thanks

**Comment by JinHoSo** [comment]
Thank you 🙏

**Comment by LucasPetersen** [comment]
This update reverted the last issue:  "Bugfix: Dispose"

**Pull request merged**